### PR TITLE
feat: type-check examples in the manual

### LINKED ISSUE #31: Typecheck examples in the manual
With https://github.com/denoland/deno/pull/11421 landed, we can now use `canary` channel to type check examples in the manual.

Since manual uses `$STD_VERSION` tags, we'll need to add an import map file, that will map to latest version of standard library.

This functionality should be added as part of CI script.

CC @caspervonb

### PR DESCRIPTION
**Pull request by GJZwiers**
Closes #31 

This is a redo of https://github.com/denoland/manual/pull/92 because I had to delete and recreate my fork. It includes the changes as suggested from the feedback in that PR.

### CODE REVIEW

**Comment by bartlomieju** [comment]
@GJZwiers can you add type-checking as a separate CI step instead of adding it to already existing `format` step?

**Comment by GJZwiers** [comment]
@bartlomieju done! you can view the remaining type-check errors in the pipeline output. I've also included your import map idea to map the `$STD_VERSION` placeholders to the latest `std`

**Comment by bartlomieju** [comment]
> @bartlomieju done! you can view the remaining type-check errors in the pipeline output. I've also included your import map idea to map the `$STD_VERSION` placeholders to the latest `std`

That looks good! Thank you, do you mind if I push the fixes to your PR?

**Comment by GJZwiers** [comment]
@bartlomieju Not at all! :)

**Comment by bartlomieju** [comment]
Looks like test command is broken... I will need to fix it in Deno first

File: typescript/types.md [hunk 1]
```diff
@@ -178,7 +178,7 @@ following triple-slash directives near the top of the entry point file for the
 worker script, Deno will now type check it as a Deno worker script, irrespective
 of how the module is analyzed:
 
-```ts
+```ts ignore
 /// <reference no-default-lib="true" />
 /// <reference lib="deno.worker" />
 ```
```
Comment (GJZwiers) [inline typescript/types.md:184]: @bartlomieju I did some debugging and found out that these triple-slash directives are messing up the typechecking during `deno test --doc --unstable`! After ignoring it all the errors related to unstable APIs not being found even though we're passing --unstable are gone. :)

Comment (GJZwiers) [reply typescript/types.md:184]: Still 33 errors left, but they are 'legit'. I will try and fix some of them in the coming days

**Comment by bartlomieju** [comment]
Hey @GJZwiers, sorry I still haven't been able to fix underlying issue in Deno. Maybe you want to take a stab at it? I'd be glad to provide guidance: https://github.com/denoland/deno/issues/12861

**Comment by GJZwiers** [comment]
> Hey @GJZwiers, sorry I still haven't been able to fix underlying issue in Deno. Maybe you want to take a stab at it? I'd be glad to provide guidance: [denoland/deno#12861](https://github.com/denoland/deno/issues/12861)

No worries, I believe the problem has been fixed, see my comment [here](https://github.com/denoland/deno/issues/12861#issuecomment-1062776243). This PR is pretty much ready to land now, my only remaining concern is that the syntax highlighting is not working for code blocks with an `ignore` attribute in the preview deployments.

**Comment by GJZwiers** [comment]
@bartlomieju This thing is finally done! A final review might be good though

**Comment by bartlomieju** [comment]
> @bartlomieju This thing is finally done! A final review might be good though

Thanks! I'll take a look. Is the issue with clashing types somehow resolved now?

**Comment by GJZwiers** [comment]
> Thanks! I'll take a look. Is the issue with clashing types somehow resolved now?

Yes, it was due to [a block with triple-slash directives](https://github.com/denoland/manual/blob/main/typescript/types.md?plain=1#L181) that caused a log of subsequent blocks to be type-checked as worker scripts. Excluding the block suffices for the manual, If more action is needed on it in the future, there is still [the issue tracker in denoland/deno](https://github.com/denoland/deno/issues/12861).

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @GJZwiers! Massive effort

**Pull request merged**